### ISSUE TITLE
Ag-grid flickers when scrolling to bottom when using React 18 

### ISSUE THREAD

**Issue by camposfyi** [ag-grid/ag-grid#5647]
I'm submitting a:   
```
[X] bug report
```

### Set up:
```
* ag-grid 28.2.0
* react 18.2.0
* Chrome: 106.0
* Windows 10 Pro
```

### Issue:
When scrolling to the bottom of the grid, the rows disappear briefly causing a flicking effect.

In my use case I am inserting hundreds of rows per second. I am using **applyTransactionAsync** with **asyncTransactionWaitMillis={100}** as a prop in the grid, then when **onRowDataUpdated** is fired I scroll to the last row.
```
            const lastRowIndex = agGridApi.getDisplayedRowCount() - 1;
            if (lastRowIndex > 0) {
                agGridApi.ensureIndexVisible(lastRowIndex, 'bottom');
            }
```

Note that the issue only occurs when using **createRoot.render** from **react-dom/client** which is the new way to render your app in React 18.

The reason why it's important that I use React 18 is because of the timing it takes to scroll to the bottom with my data set. Using react 18 it takes 3ms to scroll to the bottom, and using React 17 takes 30ms, that's 10X longer and I need to scroll the grid very often.

I have two plunkers that are identical, except that one renders the application using **createRoot.render** from react-dom/client (this one flickers), and a second one that renders the application using **ReactDOM.render** (this one does not flicker)

Grid **flickers** (createRoot.render):
https://plnkr.co/plunk/enauSEO1zDAz3QYH

Grid **does not flicker** (ReactDOM.render)
https://plnkr.co/plunk/jVcD25Lxr1rEJUsV

### Note:
I've noticed that when the grid flickers and the rows disappear they are positioned way up high in the **ag-center-cols-container** element. They are on the DOM they're just not positioned correctly.

**Comment by kiril-matev** [comment]
Hi @camposfyi,

Thank you for bringing this up.

I've now updated this ticket metadata and it should show up in the pipeline in a couple of hours.

Thank you once again!

**Comment by camposfyi** [comment]
I see it now. Closing this ticket.

**Comment by camposfyi** [comment]
@kiril-matev I'm going to look into fixing this myself. Do you have a guide on how to set up ag-grid local development? I'm not finding it straight forward from the package.json and various readme files in the source code. Thanks.

**Comment by kiril-matev** [comment]
Hi @camposfyi,

Thank you for your willingness to help resolve this!

However, we don't have a getting started guide on local development. I would recommend holding off from spending any time on investigating this as you'll need to do a lot of testing to ensure your fix will work in all cases and not just in the particular case you're investigating and also it doesn't affect the performance of the grid. 

We'll be looking into a range of issues related to React 18 and we hope to be able to address them soon. This is why I'd suggest leaving this to us to fix as we're aware of the issue and are keen to resolve it.

Thanks for your time - we appreciate it!

**Comment by uvtzxpm** [comment]
@StephenCooper Is there any update on this? I'm having the same problem.

There's been a new ag-grid version since this issue was created, but AG-7429 is still in the [pipeline](https://www.ag-grid.com/pipeline/) with status "Backlog". It's still broken with v29.0.0.

**Comment by kiril-matev** [comment]
@uvtzxpm We're planning to work on this item soon and we will try to fix it in our next release.

In the meantime, you can avoid this issue as described in the bug report description above by using **ReactDOM.render** instead.

**The best way to get an update on this item is to sign up for AG Grid new release notifications. Once a new AG Grid version is released, look for the item reference above (AG-XXXX) on the changelog page to see if it's resolved in that version:
https://www.ag-grid.com/ag-grid-changelog/**

**Comment by kiril-matev** [comment]
Hi @camposfyi,

We will fix this bug in our next release - v29.2 to be released before March 24th.

In order to fix this, you'll also need to:
**1/** update the code which does scrolling in the onRowDataUpdated event handler. Use the code below:
     flushSync(()=>{
        this.gridApi.ensureIndexVisible(toRow++, 'bottom');
     })

**2/** set **suppressAnimationFrame: true**

**These code changes are part of the sample below - this sample still uses v29.1 so it has the issue you reported. Once v29.2 is released, please upgrade it to v29.2 to see this fixed.**
https://plnkr.co/edit/dabAb4ndpljSP6Ck

Thanks for your time - we appreciate it!
